src/snowflake/connector/result_batch.py

```diff
@@ -72,21 +72,25 @@ def create_batches_from_response(
     cursor: "SnowflakeCursor",
     _format: str,
     data: Dict[str, Any],
+    schema: List["ResultMetadata"],
 ) -> List["ResultBatch"]:
     column_converters: List[Tuple[str, "SnowflakeConverterType"]] = []
     arrow_context: Optional["ArrowConverterContext"] = None
     rowtypes = data["rowtype"]
-    column_names: List[str] = [c["name"] for c in rowtypes]
     total_len: int = data.get("total", 0)
     first_chunk_len = total_len
     rest_of_chunks: List["ResultBatch"] = []
     if _format == "json":
-        column_converters: List[Tuple[str, "SnowflakeConverterType"]] = [
-            (
-                c["type"],
-                cursor._connection.converter.to_python_method(c["type"].upper(), c),
+
+        def col_to_converter(col):
+            type_name = col["type"].upper()
+            python_method = cursor._connection.converter.to_python_method(
+                type_name, col
             )
```

Comment:
Added a small function here so that we can name the two values in the tuple (type name and python method)